### PR TITLE
[EPMEDU-871] upgrade Gradle to 7.3.0 with dependencies

### PR DESCRIPTION
**Pull request by aklepets**
- Upgrade Gradle version to 7.3.0
- upgrade dependent libraries

### CODE REVIEW

**Review by Makentoshe — APPROVED** [review]

**Comment by egorikftp** [comment]
@aklepets  Let's also migrate to `namespace`
https://developer.android.com/studio/releases/gradle-plugin#package-deprecated

Android Studio Upgrade tool should do it automatically
Also please apply other suggestions if available

Also please remove + dependency from `libs.versions.toml`
image

`library/foundation/build.gradle.kts:16`

**Review by Fileek — APPROVED** [review]

**Review by n3gbx — APPROVED** [review]

File: app/build.gradle.kts [hunk 1]
```diff
@@ -15,6 +15,7 @@ animealPlugin {
 }
 
 android {
+    namespace = "com.epmedu.animeal"
```
Comment (egorikftp) [inline app/build.gradle.kts:18]: `namespace` should be in all modules and need to remove `package` from AndroidManifest

File: gradle/libs.versions.toml [hunk 2]
```diff
@@ -1,31 +1,31 @@
...
-compileSdk = "32"
-targetSdk = "32"
+compileSdk = "33"
+targetSdk = "33"
 minSdk = "24"
 
 accompanist = "0.25.1"
-activityCompose = "1.5.1"
-amplify = "1.37.1"
+activityCompose = "1.6.0-rc02"
```
Comment (egorikftp) [inline gradle/libs.versions.toml]: As I remember we agreed don't use RC dependencies and modify filtering logic (remove "RC")
https://github.com/AnimealProject/animeal_android/blob/bc88312b1b7fe36fda389c30c61abe2b9ac11da3/build.gradle.kts#L72

File: library/foundation/build.gradle.kts [hunk 3]
```diff
@@ -15,7 +15,5 @@ dependencies {
     implementation(libs.compose.material.dialog.datetime)
 
     // https://developer.android.com/studio/known-issues#error_when_rendering_compose_preview
-    debugImplementation(libs.androidx.lifecycle.runtime)
-    debugImplementation(libs.androidx.lifecycle.savedstate)
```
Comment (egorikftp) [inline library/foundation/build.gradle.kts:19]: `savedstate` unused and can be removed from `libs.versions.toml`

File: library/foundation/build.gradle.kts [hunk 4]
```diff
@@ -15,7 +19,5 @@ dependencies {
     implementation(libs.compose.material.dialog.datetime)
 
     // https://developer.android.com/studio/known-issues#error_when_rendering_compose_preview
```
Comment (egorikftp) [inline library/foundation/build.gradle.kts]: unnecessary commentary

**Review by egorikftp — APPROVED** [review]

File: library/foundation/build.gradle.kts (hunk 4 above)
Comment (egorikftp) [reply library/foundation/build.gradle.kts]: removed

**Pull request merged**